### ISSUE TITLE
iRods microservice does not catch ibun (zip/unzip) actions

### ISSUE THREAD

**Issue by devincowan** [hydroshare/hydroshare#5418]
we've noticed that in some cases the size displayed on resource landing page differs vastly from that in iRods

Example:

- https://www.hydroshare.org/resource/62dfabfc99c64878883e58383182dd92/
- iRods AVU shows: ~325.94 GB in the datazone
- Django [FileField sizes](https://docs.djangoproject.com/en/3.2/ref/files/file/#django.core.files.File.size) calculate the size of the resource at 135.6GB

![image](https://github.com/hydroshare/hydroshare/assets/17934193/8b1e9c53-dd8e-4dbb-9160-c60189bf8c4f)

```shell
$ ils ../wwwHydroProxy/62dfabfc99c64878883e58383182dd92
/hydroshareZone/home/wwwHydroProxy/62dfabfc99c64878883e58383182dd92:
  C- /hydroshareZone/home/wwwHydroProxy/62dfabfc99c64878883e58383182dd92/data
$ # quotaUserName AVU on the collection
$  imeta ls -C ../wwwHydroProxy/62dfabfc99c64878883e58383182dd92 quotaUserName
AVUs defined for collection /hydroshareZone/home/wwwHydroProxy/62dfabfc99c64878883e58383182dd92:
attribute: quotaUserName
value: REDACTED
units:
$ # quota usage for the quotaUserName (AVU named is quotaUserName-usage)
$ imeta ls -C ../wwwHydroProxy/bags REDACTED-usage
AVUs defined for collection /hydroshareZone/home/wwwHydroProxy/bags:
attribute: REDACTED-usage
value: 349970851238

? # actual aggregate size on disk of the collection minus the FAIR data objects
? # but INCLUDING the overhead of all the subdirectory book-keeping on disk (approx 4K per subdir)
? du -b --max-depth=0 /projects/hydroshare/vaults/wwwProxy/home/wwwHydroProxy/62dfabfc99c64878883e58383182dd92/data/contents
349971928486	/projects/hydroshare/vaults/wwwProxy/home/wwwHydroProxy/62dfabfc99c64878883e58383182dd92/data/contents
```

Perhaps related to https://github.com/hydroshare/hydroshare/issues/5182

_Originally posted by @devincowan in https://github.com/hydroshare/hydroshare/issues/5262#issuecomment-1989117821_

**Comment by devincowan** [comment]
UPDATE: after repairing the resource and ingesting all irods files, then running command to refresh size metadata, I see:
```python
>>> from hs_core.models import BaseResource
>>> res = BaseResource.objects.get(short_id="62dfabfc99c64878883e58383182dd92")
>>> res.size
315937358286
>>> from hs_core.hydroshare.utils import convert_file_size_to_unit
>>> convert_file_size_to_unit(res.size, 'gb')
294.2395939361304
>>> from hs_tools_resource.utils import convert_size
>>> convert_size(res.size)
'294.24 GB'
>>> from django.template.defaultfilters import filesizeformat
>>> filesizeformat(res.size)
'294.2\xa0GB'
```
We might see quite a bit of variation depending on how we convert size
For the resource landing page, we use the built-in: https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#filesizeformat
https://github.com/django/django/blob/3.2.11/django/template/defaultfilters.py#L838-L877

**Comment by devincowan** [comment]
Other issues
https://gist.github.com/devincowan/40f29b5f22d126d9e4db5d6585f0dcb5

**Comment by devincowan** [comment]
closing as no longer needed
Quota [updates released as 2.14](https://github.com/hydroshare/hydroshare/issues/5228) move the HS datazone away from microservices.
Improper handling of `ibun` should not be impactful for the userzone (where microservices will still be enabled for the time being)
